api-contract/contract.yaml

```diff
@@ -348,7 +352,167 @@ paths:
         default:
           description: successful operation
           content: {}
-          
+  /friend:
+    post:
+      tags:
+      - friend
+      summary: Add a friend
+      description: This can be done by the logged in user.
+      operationId: addFriend
+      requestBody:
+        description: Add friend object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Friend'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
+    delete:
+      tags:
+      - friend
+      summary: Delete a friend
+      description: This can be done by the logged in user.
+      operationId: deleteFriend
+      requestBody:
+        description: Delete friend object
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Friend'
+        required: true
+      security:
+        - bearer_auth: []
+      responses:
+        default:
+          description: successful operation
+          content: {}
```

Comment:
Something I've just noticed, when we delete a friend we only need two things:
* the username of the logged in user -> this can be passed through params
* the username of the friend to be removed from the list